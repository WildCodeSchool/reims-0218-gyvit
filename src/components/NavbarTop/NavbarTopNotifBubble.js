//TODO: create component notif bubble icon

import React from "react"
import PropTypes from "prop-types"
import { Badge } from "reactstrap"

const NavbarTopNotifBubble = () => {
  return (
    <Badge
      style={{
        marginTop: "-50px",
        backgroundColor: "#ff007e",
        width: "auto",
        height: "20px"
      }}
      pill
    >
      8
    </Badge>
  )
}

NavbarTopNotifBubble.propTypes = {
  notifsCount: PropTypes.number.isRequired
}

export default NavbarTopNotifBubble
