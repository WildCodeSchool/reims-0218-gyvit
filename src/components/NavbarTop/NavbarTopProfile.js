//TODO: Create component Profile

import React from "react"
import { Media } from "reactstrap"
import PropTypes from "prop-types"
import "bootstrap/dist/css/bootstrap.min.css"

const NavbarTopProfile = ({ firstname, lastname, business }) => (
  <div>
    <Media>
      <Media href="#">
        <Media
          style={{
            marginTop: "25px"
          }}
          className="rounded-circle"
          object
          src="http://via.placeholder.com/40x40"
          alt={firstname}
        />
      </Media>
      <Media body>
        <Media
          style={{
            color: "#231b56",
            fontSize: "16px",
            marginTop: "28px",
            marginLeft: "21px"
          }}
          heading
        >
          <span>{firstname}</span> <span>{lastname}</span>
        </Media>

        <Media
          style={{
            color: "#372c78",
            marginTop: "5px",
            marginLeft: "21px",
            fontSize: "14px"
          }}
        >
          {business}
        </Media>
      </Media>
    </Media>
  </div>
)

NavbarTopProfile.propTypes = {
  lastname: PropTypes.string.isRequired,
  firstname: PropTypes.string.isRequired,
  business: PropTypes.string.isRequired
}

export default NavbarTopProfile
