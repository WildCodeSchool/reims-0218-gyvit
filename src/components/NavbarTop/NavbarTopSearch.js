import React from "react"
import {
  FormGroup,
  Input,
  DropdownMenu,
  DropdownItem,
  Dropdown,
  DropdownToggle
} from "reactstrap"

//function to split with search criteria
const highlight = (text, search) => {
  const highlightSuggestion = text.toLowerCase().split(search.toLowerCase())
  return highlightSuggestion
}

const NavbarTopSearch = ({
  _id,
  onDirclick,
  searchResults,
  search,
  onSearchType
}) => (
  <div>
    <FormGroup>
      <Input
        className="typeahead form-control"
        style={{
          marginTop: "22px",
          marginLeft: "6px",
          height: "46px",
          backgroundColor: "#fbfcfd",
          borderRadius: "2px",
          border: "none"
        }}
        type="search"
        id="exampleSearch"
        placeholder="Search..."
        onChange={onSearchType}
        searchResults={searchResults}
        value={search}
      />

      <Dropdown isOpen={search}>
        <DropdownMenu
          style={{
            transform: "none",
            marginLeft: "5px",
            width: "100%"
          }}
        >
          <DropdownToggle style={{ visibility: "hidden" }} />
          {search !== "" &&
            searchResults.map(searchResult => {
              // transform a string to display in several spans
              const searchResultString = highlight(searchResult.name, search)
              return (
                <DropdownItem
                  onClick={() => onDirclick(searchResult._id)}
                  key={searchResult.value}
                  value={searchResult.value}
                >
                  <span>{searchResultString[0]}</span>
                  <span style={{ color: "red" }}>{search}</span>
                  <span>{searchResultString[1]}</span>
                </DropdownItem>
              )
            })}
        </DropdownMenu>
      </Dropdown>
    </FormGroup>
  </div>
)

export default NavbarTopSearch
