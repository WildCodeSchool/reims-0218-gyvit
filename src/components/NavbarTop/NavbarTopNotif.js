//TODO: create component notif

import React from "react"
import PropTypes from "prop-types"
import NavbarTopNotifBubble from "./NavbarTopNotifBubble"

const NavbarTopNotif = () => {
  return (
    <div>
      <img
        style={{
          marginTop: "37px"
        }}
        src={process.env.PUBLIC_URL + "img/icon_notification.png"}
        alt="iconNotif"
      />
      <NavbarTopNotifBubble notifsCount={66} />
    </div>
  )
}

NavbarTopNotif.propTypes = {
  notifsCount: PropTypes.number.isRequired
}

export default NavbarTopNotif
