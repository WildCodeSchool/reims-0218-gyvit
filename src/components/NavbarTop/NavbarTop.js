//TODO: create navbar top with search bar, notification icon and profile.

import React from "react"
import { Row, Col, Container } from "reactstrap"
// import PropTypes from "prop-types"
import NavbarTopNotif from "./NavbarTopNotif"
import NavbarTopProfileContainer from "../../containers/NavbarTopProfileContainer"
import SearchContainer from "../../containers/SearchContainer"

const NavbarTop = () => (
  <Container>
    <Row
      style={{
        height: "90px",
        backgroundColor: "#ffffff",
        border: "solid 1px #E5E8F4"
      }}
    >
      <Col xs="7">
        <SearchContainer />
      </Col>
      <NavbarTopNotif />
      <Col xs="3">
        <NavbarTopProfileContainer />
      </Col>
    </Row>
  </Container>
)

export default NavbarTop
