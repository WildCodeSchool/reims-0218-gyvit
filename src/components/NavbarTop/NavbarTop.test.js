//TODO: Test navbar top with search bar, notification icon and profile.

import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import NavbarTop from "./NavbarTop"

const props = {
  notif: {
    notifsCount: 6
  },
  profile: {
    lastname: "Duflot",
    firstname: "Romain",
    business: "Notoriety"
  }
}

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<NavbarTop {...props} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders correctly", () => {
  const tree = renderer.create(<NavbarTop {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
