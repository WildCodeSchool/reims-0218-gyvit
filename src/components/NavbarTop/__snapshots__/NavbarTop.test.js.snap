// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "border": "solid 1px #E5E8F4",
        "height": "90px",
      }
    }
  >
    <div
      className="col-7"
    >
      <div>
        <div
          className="form-group"
        >
          <span
            className="fa fa-search"
          />
          <input
            className="form-control"
            id="exampleSearch"
            name="search"
            onChange={undefined}
            placeholder="Search..."
            style={
              Object {
                "backgroundColor": "#fbfcfd",
                "border": "none",
                "borderRadius": "2px",
                "height": "46px",
                "marginLeft": "6px",
                "marginTop": "22px",
              }
            }
            type="search"
            value={undefined}
          />
        </div>
      </div>
    </div>
    <div>
      <img
        alt="iconNotif"
        src="img/icon_notification.png"
        style={
          Object {
            "marginTop": "37px",
          }
        }
      />
      <span
        className="badge badge-secondary badge-pill"
        style={
          Object {
            "backgroundColor": "#ff007e",
            "height": "20px",
            "marginTop": "-50px",
            "width": "auto",
          }
        }
      >
        8
      </span>
    </div>
    <div
      className="col-3"
    >
      <div>
        <div
          className="media"
        >
          <div
            className="media"
            href="#"
          >
            <img
              alt="Pierre"
              className="rounded-circle media-object"
              src="img/kevinMarlot.jpeg"
              style={
                Object {
                  "height": "40px",
                  "marginTop": "25px",
                  "width": "40px",
                }
              }
            />
          </div>
          <div
            className="media-body"
          >
            <h4
              className="media-heading"
              style={
                Object {
                  "color": "#231b56",
                  "fontSize": "16px",
                  "marginLeft": "21px",
                  "marginTop": "28px",
                }
              }
            >
              <span>
                Pierre
              </span>
               
              <span>
                Dumay
              </span>
            </h4>
            <div
              className="media"
              style={
                Object {
                  "color": "#372c78",
                  "fontSize": "14px",
                  "marginLeft": "21px",
                  "marginTop": "5px",
                }
              }
            >
              Wild Code School
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
